File: programmers/etc/level2/더_맵게.py
from heapq import *

def solution(scoville, K):
    answer = 0
    q = []
    for s in scoville:
        heappush(q, s)
        
    while q[0] < K:
        if len(q) <= 1:
            return -1
        a = heappop(q)
        b = heappop(q)
        heappush(q, a+b*2)
        answer += 1
        
    return answer


scoville = [1, 2, 3, 9, 10, 12]
K = 7
print(solution(scoville, K)) # 2

File: Java/programmers/etc/level2/더맵게.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 더맵게 {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> q = new PriorityQueue<>();
        
        for (int s : scoville) q.add(s);

        while (q.peek() < K) {
            if (q.size() <= 1) return -1;
            int a = q.poll(), b = q.poll();
            q.add(a+b*2);
            answer++;
        }
        
        return answer;
    }
}
